src/txmempool.cpp

```diff
@@ -1004,7 +993,7 @@ bool CCoinsViewMemPool::GetSidechain(const uint256& scId, CSidechain& info) cons
         const uint256& scCreationHash = mempool.mapSidechains.at(scId).scCreationTxHash;
         const CTransaction & scCreationTx = mempool.mapTx.at(scCreationHash).GetTx();
         for (const auto& scCreation : scCreationTx.GetVscCcOut()) {
-            if (scId == scCreation.scId) {
+            if (scId == scCreation.GetScId()) {
```

Comment:
IMPORTANT: GetSidechain is NOT working properly if in the mempool there are two transactions creating two different scs. In particular if there is TX1 creating SC1 and TX2 creating SC2, GertSidechain(2) will return an empty SCInfo! Move the break at line 1005 inside the if.

Reply:
> IMPORTANT: GetSidechain is NOT working properly if in the mempool there are two transactions creating two different scs. In particular if there is TX1 creating SC1 and TX2 creating SC2, GertSidechain(2) will return an empty SCInfo! Move the break at line 1005 inside the if.

SOLVED